core/templates/pages/contributor-dashboard-page/services/contribution-and-review-backend-api.service.spec.ts

```diff
@@ -140,7 +140,9 @@ describe('Contribution and review backend API service', () => {
           'add_question',
           AppConstants.OPPORTUNITIES_PAGE_SIZE,
           0,
-          'Date'
+          'Date',
+          undefined,
+          undefined
```

Comment:
Should `undefined` be passed here explicitly?
For example:
https://github.com/oppia/oppia/blob/556edf2dda33a4ff16c1be630a4f42dc612324a1/core/templates/pages/contributor-dashboard-page/services/contribution-and-review-backend-api.service.spec.ts#L168-L176

here we don't pass a `undefined` value for `topicName`

Reply:
Just a note: generally speaking, for TS, if the semantic we're going for is `null`, it's better to pass null rather than undefined.

Reply:
Thanks for the suggestion. I did a few tests with `null`, and even removing both `undefined` from the test code. However, even though I understand what both of you mean,  actually the semantic here is correct: `undefined`. This is because:
1. The function can be called with a variable number of parameters. When called with 2 parameters less, the `toHaveBeenCalledWith()` function fills the missing parameters with `undefined` (so, removing them here is not an option)
2. In `fetchReviewableSuggestionsAsync`, these 2 parameters (`explorationId?` and `topicName?`) are receiving its values from places where the value may be `undefined` (because these other functions have `explorationId?` and `topicName?` in turn), so it makes sense that sometimes the `fetchReviewableSuggestionsAsync` function will be called passing a variable with `undefined` value.

This can also be seen in other places of the existing code:

<img width="600" alt="Screenshot 2023-04-18 at 12 18 10 AM" src="https://user-images.githubusercontent.com/2841740/232702842-c9800d96-2e62-4f3c-b751-d9523f820da4.png">

Reply:
@jahd2602 Thanks for the explanation! I agree with the (implicit) use of undefined for the case of variadic functions (where some args are omitted at the end), and I think that's OK.

But if we pass optional arguments explicitly by calling a function using those arguments, I think we should be using `null` (and consequently I think that the other places where we're passing `undefined` explicitly in the codebase are actually not correct). That's because `null` is typically used when the absence of the value is intentional (as it would be, if we were explicitly making a call with it), whereas `undefined` is used when the variable may not even exist (e.g. a non-existent value in a JavaScript "dictionary"). So in the case above I think the call to carbas.fetchSuggestionsAsync() should be using null since that's a value that explicitly tells the called function that the request doesn't correspond to an exploration ID.

Whether or not to fix the existing cases in this PR is a separate issue (and is up to you), but I suggest not propagating the practice, if possible.

Reply:
Found other instances inside the file where `undefined` is used. Added a follow up issue https://github.com/oppia/oppia/issues/18113 so that we can get this PR out the door

Reply:
@jahd2602 Can you add the issue as a todo comment in the file?

Reply:
Good idea! Added, thanks!